### PR TITLE
Clean up the build configuration and change project description.

### PR DESCRIPTION
**Pull request by Hjdskes**
Hello,

This PR changes two general things:

1. It changes the project description to something that is more complete, and in my opinion also something more professional which suits a popular project better.
2. It applies the use of [git.mk](https://github.com/behdad/git.mk) to the build process, automating the generation of `.gitignore` files. Whilst working on this, I also added `COPYING` and `README.md` to `EXTRA_DIST`, meaning that these two files are now also distributed. I also removed some unused things from `configure.ac`.

I appreciate that changing the problem description may be opinionated, so feel free to only cherry-pick the second commit. I have some more changes coming, but those affect the code and will be in separate PRs. It'll be a while before I open those, as it has been a few months since I worked on them and would like to look them over and properly test them. If you want to take an early look, the changes can be found in the feature branches in my forked repository.

Thanks,

Jente

### CODE REVIEW

**Comment by acrisci** [comment]
Thanks for taking a look at the project. I'll review this as soon as I get some time.

**Comment by acrisci** [comment]
First of all, I kind of like the project name how it is. It comes from the song Notorious BIG - Big Papa ("throw your hands in the air if you're a true player"). It has the double meaning that 1) the music players that support the spec are "true players" and also that 2) the people who use the program are "true players" as well. That's good marketing because it suggests that users of this software are in among the "cool crowd".

Also, we need to keep the list of software it supports in the tag line for SEO purposes. Putting MPRIS in the title maybe a little too much technical detail for people for a first impression (unless people are actually searching for MPRIS command line programs, which I don't think they are).

**Comment by acrisci** [comment]
Second of all, I'm not sold on the git.mk script either. It seems like a pretty cool idea, but I'm not exactly sure it solves a problem that the project has.

Here are some shortcomings I can think of:

1. It's difficult to add custom things to the gitignore (like a ycm complete file, IDE config, or whatever I might want to put in there for personal dev purposes).
2. It's difficult to see at a glance what is in the gitignore.
3. If anything goes wrong, I have to debug the script.
4. You have to build the project to get the gitignore (so someone making README changes will not have it for instance).

That seems like a lot of overhead, especially for a new contributor to the project.

On the other hand, the way it works now hasn't cause us problems and it follows git standards that everybody already knows.

**Comment by acrisci** [comment]
With that said, you'll have to sell me on these changes and address those points.

**Comment by Unia** [comment]
> First of all, I kind of like the project name how it is. It comes from the song Notorious BIG - Big Papa ("throw your hands in the air if you're a true player"). It has the double meaning that 1) the music players that support the spec are "true players" and also that 2) the people who use the program are "true players" as well. That's good marketing because it suggests that users of this software are in among the "cool crowd".

I did not know this -- and I guess with me are the majority of people. Suggesting users of your software are "in the cool crowd" is rather unprofessional; are people who use mpd instead of playerctl "not cool"? For those who don't know, calling only MPRIS-supporting players "true players" is condescending to those who do not support MPRIS. Neither are good marketing.

> Also, we need to keep the list of software it supports in the tag line for SEO purposes. 

You can keep a line of players supporting MPRIS, just don't put it in the description of the project. Also, when keeping this line of players, are you making sure it stays up to date? Right now, it is incomplete.

> Putting MPRIS in the title maybe a little too much technical detail for people for a first impression (unless people are actually searching for MPRIS command line programs, which I don't think they are).

I found playerctl precisely through such a search. Why would people be looking for a command line utility to control MPRIS players, if they aren't command line users in the first place?

Anyway, this is only a minor issue to me (for the reasons I just stated) and it is your project -- feel free to reject this change! ;)

---

> Second of all, I'm not sold on the git.mk script either. It seems like a pretty cool idea, but I'm not exactly sure it solves a problem that the project has.

> Here are some shortcomings I can think of:

> It's difficult to add custom things to the gitignore (like a ycm complete file, IDE config, or whatever I might want to put in there for personal dev purposes).

As stated in the header of git.mk ([here](https://github.com/behdad/git.mk/blob/master/git.mk#L26)), these files do not belong in a project's .gitignore file! See `man gitignore`.

> It's difficult to see at a glance what is in the gitignore.

This I agree with, although is this much of an issue? When is the last time you actually needed to do this?

As a packager, you don't care about what's in the .gitignore file. As a developer, you might, but then the file has been generated already and you can just open it as you would now.

> If anything goes wrong, I have to debug the script.

Valid point, but let me point out that the script is used by major projects. Most of the GNOME projects, for example, include it in their repositories. I don't expect much issues here.

> You have to build the project to get the gitignore (so someone making README changes will not have it for instance).

Not entirely true; someone can simply generate only the gitignore files: https://github.com/behdad/git.mk/blob/master/git.mk#L53. I agree that this is an extra step to take.

> On the other hand, the way it works now hasn't caused us problems and it follows git standards that everybody already knows.

Valid point, although using git.mk does offer means to improve your usage of gitignore. As I said, `man gitignore`; local development files do not belong in project's gitignore files.

---

If you want, I can extract the changes that we haven't discussed (assuming you agree with those) into a separate commit, which you can then cherry-pick without having to pull in the git.mk and project description changes. Let me know which way you want to go with this.

**Comment by acrisci** [comment]
> Suggesting users of your software are "in the cool crowd" is rather unprofessional

Remember [these famous commercials](https://www.youtube.com/watch?v=qfv6Ah_MVJU) for Apple products? It's basically the same idea as that. I won't claim to be a marketing expert, but there's nothing wrong with saying people who use my software are cool B)

> You can keep a line of players supporting MPRIS, just don't put it in the description of the project.

People find the software by searching for keywords. Popular keywords should go in the description so more people see it. The names of the players (I think) are very popular keywords. Without the players in the description, the project will be less visible.

**Comment by acrisci** [comment]
As for the git.mk stuff, I might warm to that so I'll think about it some more and make a decision later.

**Comment by Unia** [comment]
Alright, you won't accept d9bca2dceb0c13595d207ca036c4ef1020c07361. Have you thought more about the other commit?

**Comment by acrisci** [comment]
So the git.mk stuff I don't really like. It just seems like a solution in search of a problem to me. It also violates [the principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) because it changes how git works in a way that people do not expect.

As for the project name, yes it's kind of stupid, but it's a fun kind of stupid. I'm not convinced a dryer name will be more attractive to potential users.

So for those reasons, I am going to close this. But thanks for your contribution to the project by putting these ideas up for consideration.

File: Makefile.am [hunk 1]
```diff
@@ -1,3 +1,15 @@
 ACLOCAL_AMFLAGS = -I build/autotools
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc --enable-introspection
 SUBDIRS=playerctl doc
+
+EXTRA_DIST = \
```
Comment (Hjdskes) [inline Makefile.am]: I would advice you to get these two lines into master: *Still, sometimes there are files that must be distributed, but which are not covered in the automatic rules. These files should be listed in the EXTRA_DIST variable. You can mention files from subdirectories in EXTRA_DIST.*[source](https://www.gnu.org/software/automake/manual/html_node/Basics-of-Distribution.html).

File: configure.ac [hunk 2]
```diff
@@ -15,30 +15,21 @@ AC_CONFIG_HEADER([config.h])
```
Comment (Hjdskes) [inline configure.ac]: The changes here are unrelated to `git.mk` and clean up the build. I would advice you to cherry-pick these changes as well.
